cpp/src/arrow/array/validate.cc

```diff
@@ -413,6 +414,20 @@ struct ValidateArrayImpl {
     return Status::OK();
   }
 
+  Status Visit(const RunEndEncodedType& type) {
+    switch (type.run_ends_type()->id()) {
+      case Type::INT16:
+        return ValidateRunEndEncoded<int16_t>(type);
+      case Type::INT32:
+        return ValidateRunEndEncoded<int32_t>(type);
+      case Type::INT64:
+        return ValidateRunEndEncoded<int64_t>(type);
+      default:
+        return Status::Invalid("Run ends type must be int16, int32 or int64, but got: ",
+                               type.run_ends_type());
```

Comment:
Hmm, does it print out the `shared_ptr<DataType>`?

Reply:
Added a `->ToString()`